contrib/drivers/huawei/oceanstor/client.go

```diff
@@ -1011,58 +1023,93 @@ func (c *OceanStorClient) RemoveHostGroupFromMappingView(viewId, hostGrpId strin
 		log.Infof("Host group %s has already been removed from mapping view %s", hostGrpId, viewId)
 		return nil
 	}
+
 	url := "/mappingview/REMOVE_ASSOCIATE"
 	data := map[string]interface{}{
 		"ASSOCIATEOBJTYPE": ObjectTypeHostGroup,
 		"ASSOCIATEOBJID":   hostGrpId,
 		"TYPE":             ObjectTypeMappingView,
 		"ID":               viewId}
+
 	if err := c.request("PUT", url, data, nil); err != nil {
+		if c.checkErrorCode(err, ErrorHostGroupNotInMappingView) {
+			return nil
+		}
+
 		log.Errorf("Remove host group %s from mapping view %s failed", hostGrpId, viewId)
 		return err
 	}
+
 	log.Infof("Remove host group %s from mapping view %s success", hostGrpId, viewId)
 	return nil
 }
 
 func (c *OceanStorClient) RemoveHostFromHostGroup(hostGrpId, hostId string) error {
-
 	url := fmt.Sprintf("/host/associate?TYPE=14&ID=%s&ASSOCIATEOBJTYPE=21&ASSOCIATEOBJID=%s",
 		hostGrpId, hostId)
 	if err := c.request("DELETE", url, nil, nil); err != nil {
+		if c.checkErrorCode(err, ErrorHostNotInHostGroup) {
+			return nil
+		}
+
 		log.Errorf("Remove host %s from host group %s failed", hostId, hostGrpId)
 		return err
 	}
+
 	log.Infof("Remove host %s from host group %s success", hostId, hostGrpId)
 	return nil
 }
 
 func (c *OceanStorClient) RemoveIscsiFromHost(initiator string) error {
-
 	url := "/iscsi_initiator/remove_iscsi_from_host"
 	data := map[string]interface{}{"TYPE": ObjectTypeIscsiInitiator, "ID": initiator}
 	if err := c.request("PUT", url, data, nil); err != nil {
+		if c.checkErrorCode(err, ErrorInitiatorNotInHost) {
+			return nil
+		}
+
 		log.Errorf("Remove initiator %s failed", initiator)
 		return err
 	}
+
 	log.Infof("Remove initiator %s success", initiator)
 	return nil
 }
 
 func (c *OceanStorClient) DeleteHostGroup(id string) error {
-	return c.request("DELETE", "/hostgroup/"+id, nil, nil)
+	err := c.request("DELETE", "/hostgroup/"+id, nil, nil)
+	if err != nil && c.checkErrorCode(err, ErrorHostGroupNotExist) {
+		return nil
+	}
+
+	return err
```

Comment:
If we have caught ErrorHostGroupNotExist, then we will return nil but if there is some other error we return err. Will that be consistent for the caller?

Reply:
Yes, we return err as before to keep consistent for other errors.